pkg/chunk/storage/factory.go

```diff
@@ -69,3 +71,24 @@ func NewTableClient(cfg Config) (chunk.TableClient, error) {
 		return nil, fmt.Errorf("Unrecognized storage client %v, choose one of: aws, gcp, inmemory", cfg.StorageClient)
 	}
 }
+
+// Opts returns the StorageOpts for the given configuration.
+func Opts(cfg Config, schemaCfg chunk.SchemaConfig) ([]chunk.StorageOpt, error) {
```

Comment:
Last comment (I promise!) - I think `NewStorageClient` should return `[]chunk.StorageOpt` now, the current way around seems weird.